File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    var stack=[]
    var x=0
    if (tokens.length<2){
        return tokens[0]
    }else{
        for(var i=0; i< tokens.length; i++){
         if (tokens[i]=='+' || tokens[i]=='-' || tokens[i]=='*' || tokens[i]=='/'){
           x= Math.trunc(eval('(' +stack[stack.length-2] + ')' + ''  +tokens[i] + ''+ '('+stack[stack.length-1]+')' ))
           stack.pop()
            stack.pop()
            stack.push(x)
        }else{
            stack.push(tokens[i])
        }
    }
    }
    
    return (x)
};

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack= []
        ops= {"+", "-", "/", "*"}
        
        for token in tokens:
            val = token
            if token in ops:
                val1= stack.pop()
                val2= stack.pop()
                val= int(eval("".join([val2, token, val1])))
            stack.append(str(val))
            
        return int(stack[0])